### PR TITLE
for local use, recommend Bundler and provide config

### PR DESCRIPTION
**Pull request by SethTisue**
so that people can test the site locally without having to
install stuff systemwide on their local machines

based on an earlier version originally contributed by Simon Ochsenreither
as part of a larger PR

the choice of Jekyll 1.5.1, for now, is based on Fabien's comments at
https://github.com/scala/scala-lang/pull/361

### CODE REVIEW

**Comment by SethTisue** [comment]
this is not yet ready for merge. @fsalvi, @heathermiller, can you help me make any needed further changes?

as already discussed at https://github.com/scala/scala-lang/pull/361, the Gemfile now specifies Jekyll 1.5.1, which is the Jekyll version now in use in production. and I no longer exclude `vendor` in _config.yml, since Fabien reports that was interfering with `resources/js/vendor` (and we can't be more specific for now because of _config.yml)

but the Gemfile.lock in this PR only mentions maruku, not kramdown, and at 361 @fsalvi says kramdown is the one we're using in production. where do I look on chara to see what you did to the production setup there, so I can try to update this PR to match it?

**Comment by SethTisue** [comment]
I see from https://scala-webapps.epfl.ch/jenkins/job/production_scala-lang.org-builder/371/console that it's webserver-production, not chara, that I would need to look at, but I don't have a login there.

**Comment by fsalvi** [comment]
Yes, it's not on chara but on another server.
It shouldn't be hard to reproduce the config, here's the list of gems:

*** LOCAL GEMS ***

blankslate (2.1.2.4)
classifier (1.3.4)
colorator (0.1)
commander (4.1.6)
fast-stemmer (1.0.2)
ffi (1.9.10)
highline (1.6.21)
jekyll (1.5.1)
json (1.8.3)
kramdown (1.8.0)
liquid (2.6.3, 2.5.5)
listen (1.3.1)
maruku (0.7.0)
mercenary (0.3.5)
parslet (1.5.0)
posix-spawn (0.3.11)
pygments.rb (0.6.3, 0.5.4)
rb-fsevent (0.9.5)
rb-inotify (0.9.5)
rb-kqueue (0.2.4)
redcarpet (2.3.0)
safe_yaml (1.0.4)
toml (0.1.2)
yajl-ruby (1.2.1, 1.1.0)


setting a GEM_HOME to a specific path and install jekyll (gem install jekyll -v 1.5.1) should be enough...

**Comment by fsalvi** [comment]
And I changed _config.yml to set the default renderer as Kramdown: 
https://github.com/scala/scala-lang/commit/8134b480125003f0e80c1a27b3815fe4be6d4a32

**Comment by fsalvi** [comment]
I tried to build the website with your modifications (using the webserver config) and it looks good for me, the build output is ok.

**Comment by SethTisue** [comment]
great! I tweaked the Gemfile/Gemfile.lock according to the information you provided, squashed, and pushed. thanks — delighted to have this resolved.

**Comment by heathermiller** [comment]
I had some suggestions before merging this...

File: README.md [hunk 1]
```diff
@@ -6,23 +6,35 @@ It's a static site generated by [Jekyll](https://github.com/mojombo/jekyll), and
...
 ## Dependencies
 
-You'll need Jekyll installed to generate and test the site. To get it, most people can install via RubyGems:
+This site uses a Jekyll, a Ruby framework. You'll need Ruby and Bundler installed; see [Jekyll installation instructions](http://jekyllrb.com/docs/installation/) for the details.
 
-    gem install jekyll
+## Building & Viewing
 
-OSX users might have to update RubyGems:
+cd into the directory where you cloned this repository, then install the required gems with `bundle install`. This will automatically put the gems into `./vendor/bundle`.
```
Comment (heathermiller) [inline README.md]: Please make 2 sets of instructions. Those for without bundler (most people I'm surrounded by) and for those who wish to build with bundler. 
Arguably lots of people either already have Jekyll installed system-wide on their machine or don't mind installing it system-wide.

**Comment by heathermiller** [comment]
Can you please do the above?

**Comment by SethTisue** [comment]
I had to fix up the `vendor` directory situation in 66c578540f5f203a9898736558d5a74e625d951e

**Comment by SethTisue** [comment]
> Can you please do the above?

sure, working on it.

**Comment by SethTisue** [comment]
see #375

File: README.md (hunk 1 above)
Comment (SethTisue) [inline README.md]: > lots of people either already have Jekyll installed system-wide on their machine

I believe the vast majority of people who have Jekyll on their machines have a modern version such as Jekyll 2 or Jekyll 3.

**Pull request merged**